core/vtk/ttkIntegralLines/ttkIntegralLines.cpp

```diff
@@ -233,132 +116,111 @@ int ttkIntegralLines::getTrajectories(vtkDataSet *input,
+  if(!inputScalars) {
+    this->printErr("wrong scalar field.");
     return -1;
   }
-#endif
-
-  ret = getOffsets(domain);
-#ifndef TTK_ENABLE_KAMIKAZE
   // field problem
-  if(ret) {
-    cerr << "[ttkIntegralLines] Error : wrong offsets." << endl;
+  if(inputOffsets->GetDataType() != VTK_INT
+     and inputOffsets->GetDataType() != VTK_ID_TYPE) {
+    this->printErr("input offset field type not supported.");
     return -1;
   }
-
-  if(inputOffsets_->GetDataType() != VTK_INT
-     and inputOffsets_->GetDataType() != VTK_ID_TYPE) {
-    cerr << "[ttkIntegralLines] Error : input offset field type not supported."
-         << endl;
-    return -1;
-  }
-#endif
-
-  ret = getIdentifiers(seeds);
-#ifndef TTK_ENABLE_KAMIKAZE
   // field problem
-  if(ret) {
-    cerr << "[ttkIntegralLines] Error : wrong identifiers." << endl;
+  if(!inputIdentifiers) {
+    this->printErr("wrong identifiers.");
     return -1;
   }
-#endif
-
-  const SimplexId numberOfPointsInDomain = domain->GetNumberOfPoints();
-#ifndef TTK_ENABLE_KAMIKAZE
   // no points.
   if(numberOfPointsInDomain <= 0) {
-    cerr << "[ttkIntegralLines] Error : domain has no points." << endl;
+    this->printErr("domain has no points.");
     return -1;
   }
-#endif
-
-  const SimplexId numberOfPointsInSeeds = seeds->GetNumberOfPoints();
-#ifndef TTK_ENABLE_KAMIKAZE
   // no points.
   if(numberOfPointsInSeeds <= 0) {
-    cerr << "[ttkIntegralLines] Error : seeds have no points." << endl;
+    this->printErr("seeds have no points.");
     return -1;
   }
 #endif
 
   vector<vector<SimplexId>> trajectories;
 
-  integralLines_.setVertexNumber(numberOfPointsInDomain);
-  integralLines_.setSeedNumber(numberOfPointsInSeeds);
-  integralLines_.setDirection(Direction);
-  integralLines_.setInputScalarField(inputScalars_->GetVoidPointer(0));
-  integralLines_.setInputOffsets(inputOffsets_->GetVoidPointer(0));
+  this->setVertexNumber(numberOfPointsInDomain);
+  this->setSeedNumber(numberOfPointsInSeeds);
+  this->setDirection(Direction);
+  this->setInputScalarField(inputScalars->GetVoidPointer(0));
+  this->setInputOffsets(inputOffsets->GetVoidPointer(0));
 
-  integralLines_.setVertexIdentifierScalarField(
-    identifiers_->GetVoidPointer(0));
-  integralLines_.setOutputTrajectories(&trajectories);
+  this->setVertexIdentifierScalarField(inputIdentifiers->GetVoidPointer(0));
+  this->setOutputTrajectories(&trajectories);
 
-  switch(inputScalars_->GetDataType()) {
-    vtkTemplateMacro(ret = dispatch<VTK_TT>());
-  }
-#ifndef TTK_ENABLE_KAMIKAZE
-  // something wrong in baseCode
-  if(ret) {
-    cerr << "[ttkIntegralLines] IntegralLines.execute() error code : " << ret
-         << endl;
-    return -1;
-  }
-#endif
+  this->preconditionTriangulation(triangulation);
 
-  // make the vtk trajectories
-  ret = getTrajectories(domain, trajectories, output);
+  int status = 0;
+  ttkVtkTemplateMacro(
+    triangulation->getType(), inputScalars->GetDataType(),
```

Comment:
sorry to bother lutz, but I just pushed 16d57e1167b41b3e6be04a5908cc68f76778d4f6, which makes that macro more consistent with the order the templates are specified.
you can please swap around this line like this?
```
inputScalars->GetDataType(), triangulation->getType(),
```
cheer,

Reply:
I see you already have pushed this change yourself. Thanks!